src/main/java/org/jboss/ejb/protocol/remote/RemotingEJBDiscoveryProvider.java

```diff
@@ -92,15 +92,7 @@
 
     private final ConcurrentHashMap<String, URI> effectiveAuthURIs = new ConcurrentHashMap<>();
     
-    private static final long DESTINATION_RECHECK_INTERVAL =
-            AccessController.doPrivileged((PrivilegedAction<Long>) () -> {
-                String val = System.getProperty("org.jboss.ejb.client.destination-recheck-interval");
-                try {
-                    return TimeUnit.MILLISECONDS.toNanos(Long.valueOf(val));
-                } catch (NumberFormatException e) {
-                    return TimeUnit.MILLISECONDS.toNanos(5000L);
-                }
-            });
+    private static final long DESTINATION_RECHECK_INTERVAL = SystemProperties.getLong(SystemProperties.DESTINATION_RECHECK_INTERVAL, 5000L);
 
```

Comment:
the system property value is milliseconds. Need to convert to nano.

Reply:
Good catch @chengfang , thanks!